p2p/connectionpool/connectionpool_test.go

```diff
@@ -127,13 +127,20 @@ func TestRemoteConnectionWithExistingConnection(t *testing.T) {
 	addr := "1.1.1.1"
 	cPool := NewConnectionPool(n, generatePublicKey())
 
+	lowPubkey, err := cryptoBox.NewPublicKeyFromBase58("7gd5cD8ZanFaqnMHZrgUsUjDeVxMTxfpnu4gDPS69pBU")
+	assert.NoError(t, err)
+	highPubkey, err := cryptoBox.NewPublicKeyFromBase58("FABBx9LKEo9dEpQeo6GBmygoqrrC34JnzDPtz1jL6qAA")
+	assert.NoError(t, err)
+
 	// local connection has session ID < remote's session ID
-	remotePub := generatePublicKey()
-	localSessionID := []byte("110")
-	n.SetNextDialSessionID(localSessionID)
+	remotePub := highPubkey
+	localPub := lowPubkey
+
+	localSession := net.NewSessionMockWithPubkey(localPub)
+	n.SetNextDialSessionID(localSession.ID().Bytes())
 	lConn, _ := cPool.GetConnection(addr, remotePub)
 	rConn := net.NewConnectionMock(remotePub)
-	rConn.SetSession(net.NewSessionMock([]byte("111")))
+	rConn.SetSession(net.NewSessionMockWithPubkey(remotePub))
```

Comment:
Also here... changing the name of the mock is not really needed

Reply:
Not sure I get it... This is a new function I created *in addition* to the old `NewSessionMock` that takes a pubkey as input instead of a slice of bytes.

I had to adapt the old function to create a pubkey from the byte array and it seemed silly to me to convert the pubkey to bytes and then back, even if it's only in tests.